airflow/contrib/hooks/sqoop_hook.py

```diff
@@ -101,13 +100,14 @@ def Popen(self, cmd, **kwargs):
         self.log.info("Command exited with return code %s", sp.returncode)
 
         if sp.returncode:
-            raise AirflowException("Sqoop command failed: %s", ' '.join(cmd))
+            raise AirflowException("Sqoop command failed: {}".format(' '.join(self.cmd_mask_password(cmd))))
 
     def _prepare_command(self, export=False):
-        if export:
-            connection_cmd = ["sqoop", "export"]
-        else:
-            connection_cmd = ["sqoop", "import"]
+        sqoop_cmd_type = "export" if export else "import"
+        connection_cmd = ["sqoop", sqoop_cmd_type]
+
+        for key, value in self.properties.items():
```

Comment:
had to move this up or the sqoop command is incorrect and fails:

> You must supply the generic arguments -conf, -D, and so on after the tool name but before any tool-specific arguments (such as --connect)

Reply:
Good catch, thanks